### DISCUSSION TITLE
Version tflint not supported Optional object type attributes are experimental

### DISCUSSION THREAD

**Discussion by rggg73** [terraform-linters/tflint#1616] [Q&A] [answered]
Good Night,
How are you?
I hope fine. 

Sorry for bothering you, but I have all day try to behind me a configure problem with TFLINT. After searching and reading very issues, the configuration continues with the same problem.

- Using [tflint v0.35.0] https://megalinter.github.io/v5/descriptors/terraform_tflint
- MegaLinter key: [TERRAFORM_TFLINT]
- Rules config: [/builds/rggg/tf-templates/storage/.tflint.hcl]
[tflint] main.tf - ERROR - 1 error(s)
--Error detail:
Failed to load configurations; /builds/rggg/tf-templates/storage/main.tf:19,1-29: Optional object type attributes are experimental; This feature is currently an opt-in experiment, subject to change in future releases based on feedback.
Activate the feature for this module by adding module_variable_optional_attrs to the list of active experiments.:
Error: Optional object type attributes are experimental
  on /builds/rggg/tf-templates/storage/main.tf line 19, in variable "controlPolicyList":
  19: variable "controlPolicyList" {
This feature is currently an opt-in experiment, subject to change in future releases based on feedback.
Activate the feature for this module by adding module_variable_optional_attrs to the list of active experiments.

I know if I add this on my main.tf  this works: 
terraform {
  experiments = [module_variable_optional_attrs]
}
. But I know this experiments is deprecated.I think that's problem is solved on tflint -version >=0.40 and terraform > 1.3. But I would like to know if can I configure any variable on GitLab CI respectively this error to bypass or any rule present on megalinter image.

The images that I'm using are: 
nvuillam/mega-linter:v4
megalinter/megalinter:v5

These images using tflint < 0.40.

Thanks a lot in advance

Best Regards, 
Rafael Gomes

**Comment by bendrucker** [comment, marked as answer]
Ignore annotations are limited to ignoring issues, not parser errors. De-activated experiments trigger parser errors so it is not possible to ignore them.

**Reply by rggg73** [reply]
Thanks a lot, I suppossed this situation. 
You know if it's possible change the tflint version present at image using variable configuration? Or other possibilities to bypass the error?

**Reply by bendrucker** [reply]
We don't keep up with higher level tools including megalinter so I'm not sure what options you have. Seems unlikely that this would be customizable if the tools are bundled into an image. There might be a newer image available.

**Reply by rggg73** [reply]
Thanks a lot Ben. With your advice I found a new image with the tflint necessary for my project.
